src/invoker.ts

```diff
@@ -171,16 +172,47 @@ export function wrapCloudEventFunction(
 }
 
 /**
- * Wraps event function (or event function with callback) in HTTP function
- * signature.
- * @param userFunction User's function.
- * @return HTTP function which wraps the provided event function.
+ * Wraps function with a "event" / "(data, context, [callback]) =>"" signature in HTTP function.
+ * @param userFunction The user's event function.
+ * @example (data, context) =>
+ * @example (data, context, callback) =>
+ * @return {HttpFunction} HTTP function which wraps the provided event function.
  */
 export function wrapEventFunction(
   userFunction: EventFunction | EventFunctionWithCallback
 ): HttpFunction {
   return (req: express.Request, res: express.Response) => {
-    const event = req.body;
+    // Setup our (data, context) types
+    let data = req.body.data;
+    let context: CloudFunctionsContext = req.body.context;
+
+    // If CloudEvent, convert to legacy event.
+    if (isCloudEvent(req)) {
+      const legacyEvent = convertCloudEventToLegacyEvent(req.headers, req.body);
+      data = legacyEvent.data;
+      context = legacyEvent.context;
+    } else {
+      // If a really legacy event, convert it to a normal event.
+      // See the tests for more details.
+      if (!context) {
+        // Support legacy events and CloudEvents in structured content mode, with
+        // context properties represented as event top-level properties.
+        // Context is everything but data.
+        const modifiedContext = req.body;
+
+        // Clear the property before removing field so the data object
+        // is not deleted.
+        modifiedContext.data = undefined;
+        delete modifiedContext.data;
+        context = modifiedContext;
+      } else {
+        // Set data and context normally
+        data = req.body.data;
+        context = req.body.context;
```

Comment:
Looks like these variables are already initialized to exactly these values on line 186/187. Can we just remove the else block here or remove the initial values above?